tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -1197,6 +1198,16 @@ TEST_F(ValidationTest, UAVStoreMaskMatch) {
       "uav store write mask must match store value mask, write mask is 7 and store value mask is 15.");
 }
 
+TEST_F(ValidationTest, UAVStoreMaskGap) {
+  if (m_ver.SkipDxilVersion(1, 6))
+    return;
+  RewriteAssemblyCheckMsg(
+      L"..\\CodeGenHLSL\\uav_store.hlsl", "ps_6_0",
+      "i32 2, i32 2, i32 2, i32 2, i8 15)",
+      "i32 undef, i32 2, i32 undef, i32 2, i8 10)",
+      "UAV write mask must be contiguous and without gaps: .x, .xy, .xyz, or .xyzw.");
+}
```

Comment:
I don't think we need to test all 11 invalid combinations, but maybe a couple more such as all set save x (0xE) and only W set (0x8)